### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by arulk8**
App link: https://wonderful-marshmallow-017187.netlify.app/

Login and signup page

Test user:
 email: "adarshbalika@gmail.com",
 password: "adarsh@1234"

### CODE REVIEW

**Review by raghav6202 — APPROVED** [review]
The images are not displayed. The U.I looks neat
Code looks good. Files are well structured. Good to merge👍

File: src/components/pages/Home/Home.jsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    <>
+      <section className="banner__container">
+        <div className="flex-row flex-jc-se">
+          <a
+            href="./products/index.html"
+            className="banner card card__v no-underline"
+          >
+            <img
+              className="card__image"
+              src="./assets/white-headphone.jpg"
```
Comment (sashaboi) [inline src/components/pages/Home/Home.jsx:15]: seems to be some issue with these, the imges wont loafd

File: src/components/pages/Authentication/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (isLoggedIn) {
+      navigate("/");
+    }
+  }, [isLoggedIn]);
+
+  useEffect(() => {
+    setFormValid(isEmailValid && isPasswordValid);
```
Comment (sashaboi) [inline src/components/pages/Authentication/Login.jsx:28]: nice way to valiadate form interesting!

File: src/components/Header/Header.jsx [hunk 3]
```diff
@@ -1,7 +1,32 @@
...
+    sessionData: { emailData = "" },
+    actions,
+  } = useStore();
+  const inital = getInitial(emailData);
+
+  const data = isLoggedIn ? (
+    <div className="no-underline c-aux">
+      {inital}
+      {" ["}
+      <a onClick={actions.logout} className="btn__link btn__link--error">
```
Comment (sashaboi) [inline src/components/Header/Header.jsx:19]: Nice touch!

File: src/components/pages/Authentication/Login.jsx [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+          value={email}
+          placeholder="tony.stark@gmail.com"
+          autoComplete="off"
+          onInput={(e) => setEmail(e.target.value)}
+          onBlur={(e) => onEmailChange(e.target.value)}
+        />
+        {!isEmailValid && !!email && (
+          <span className="inline-block validation__msg input__error">
+            <i className="fas fa-exclamation-triangle"></i>
+            <span className="px-3">Email is invalid</span>
```
Comment (sashaboi) [inline src/components/pages/Authentication/Login.jsx:100]: this seems to bug out if i copy paste an email id, do check

File: src/components/pages/Authentication/Login.jsx [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+          Username :
+        </label>
+        <input
+          id="username"
+          className="input"
+          type="text"
+          value={email}
+          placeholder="tony.stark@gmail.com"
+          autoComplete="off"
+          onInput={(e) => setEmail(e.target.value)}
```
Comment (sashaboi) [inline src/components/pages/Authentication/Login.jsx:94]: if you write an on change here, the valiadation error glitch might go away do check it out

**Review by sashaboi — APPROVED** [review]
other than the images issue which I am sure you will solve, looks good to merge!

File: src/components/pages/Authentication/Login.jsx (hunk 4 above)
Comment (arulk8) [reply src/components/pages/Authentication/Login.jsx:100]: yeah copy paste is not working. I tried with onChange but same result. that is an open issue I am yet to figure out the solution.

File: src/components/pages/Authentication/Login.jsx (hunk 5 above)
Comment (arulk8) [reply src/components/pages/Authentication/Login.jsx:94]: Thanks for the awesome review.

**Pull request merged**